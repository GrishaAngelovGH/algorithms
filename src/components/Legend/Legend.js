import './Legend.css'

const Legend = ({ items }) => (
    <div className='item-legend text-white'>
        <div className='legend-title'>Legend</div>
        <div className='legend-scale'>
            <ul className='legend-labels'>
                {
                    items.map((v, i) => (
                        <li key={i} className='fw-bold'>
                            <span style={{ background: v.color }}></span>
                            {v.title}
                        </li>
                    ))
                }
            </ul>
        </div>
    </div>
)

export default Legend