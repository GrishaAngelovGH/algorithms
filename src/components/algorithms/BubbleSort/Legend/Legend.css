.item-legend .legend-title {
    text-align: left;
    margin-bottom: 5px;
    font-weight: bold;
    font-size: 90%;
}

.item-legend .legend-scale ul {
    margin: 0;
    margin-bottom: 5px;
    padding: 0;
    float: left;
    list-style: none;
}

.item-legend .legend-scale ul li {
    font-size: 80%;
    list-style: none;
    margin-left: 0;
    line-height: 18px;
    margin-bottom: 2px;
}

.item-legend ul.legend-labels li span {
    display: block;
    float: left;
    height: 16px;
    width: 30px;
    margin-right: 5px;
    margin-left: 0;
    border: 1px solid #999;
}

.item-legend .legend-source {
    font-size: 70%;
    color: #999;
    clear: both;
}

.item-legend a {
    color: #777;
}