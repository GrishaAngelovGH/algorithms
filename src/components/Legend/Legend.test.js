import { render } from '@testing-library/react'

import Legend from './Legend'

test('should render component', () => {
    const container = render(
        <Legend
            items={[
                { color: '#198754', title: 'Regular element' },
                { color: '#ffc107', title: 'Swapped element' }
            ]}
        />
    )

    expect(container).toMatchSnapshot()
})