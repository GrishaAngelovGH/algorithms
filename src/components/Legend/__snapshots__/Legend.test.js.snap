// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Legend component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="item-legend text-white"
      >
        <div
          class="legend-title"
        >
          Legend
        </div>
        <div
          class="legend-scale"
        >
          <ul
            class="legend-labels"
          >
            <li
              class="fw-bold"
            >
              <span
                style="background: rgb(25, 135, 84);"
              />
              Regular element
            </li>
            <li
              class="fw-bold"
            >
              <span
                style="background: rgb(255, 193, 7);"
              />
              Swapped element
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="item-legend text-white"
    >
      <div
        class="legend-title"
      >
        Legend
      </div>
      <div
        class="legend-scale"
      >
        <ul
          class="legend-labels"
        >
          <li
            class="fw-bold"
          >
            <span
              style="background: rgb(25, 135, 84);"
            />
            Regular element
          </li>
          <li
            class="fw-bold"
          >
            <span
              style="background: rgb(255, 193, 7);"
            />
            Swapped element
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
